bibliopixel/led/coord_map.py

```diff
@@ -0,0 +1,143 @@
+from .. util import pointOnCircle
+
+
+class Rotation:
+    ROTATE_0 = 0  # no rotation
+    ROTATE_90 = 3  # rotate 90 degrees
+    ROTATE_180 = 2  # rotate 180 degrees
+    ROTATE_270 = 1  # rotate 270 degrees
+
+
+def rotate_and_flip(coord_map, rotation, flip):
+    for i in range(rotation):
+        coord_map = list(zip(*coord_map[::-1]))
+
+    if flip:
+        coord_map = coord_map[::-1]
+
+    return coord_map
+
+
+def gen_matrix(dx, dy, serpentine=True, offset=0,
+               rotation=Rotation.ROTATE_0, y_flip=False):
+    """Helper method to generate X,Y coordinate maps for strips"""
+
+    result = []
+    for y in range(dy):
+        if not serpentine or y % 2 == 0:
+            result.append([(dx * y) + x + offset for x in range(dx)])
+        else:
+            result.append([((dx * (y + 1)) - 1) - x + offset for x in range(dx)])
+
+    result = rotate_and_flip(result, rotation, y_flip)
+
+    return result
+
+
+def gen_cube(dx, dy, dz, xy_serpentine=True, offset=0,
+             xy_rotation=Rotation.ROTATE_0, z_rotation=Rotation.ROTATE_0,
+             y_flip=False, z_flip=False):
+    result = []
+    plane_offset = offset
+    for z in range(dz):
+        plane = gen_matrix(dx, dy, serpentine=xy_serpentine,
+                           offset=plane_offset, rotation=xy_rotation,
+                           y_flip=y_flip)
+        plane_offset += (dx * dy)
+        result.append(plane)
+
+    result = rotate_and_flip(result, z_rotation, z_flip)
+
+    return result
+
+
+def point_list_from_cube(coord_map):
+    num = 0
+    dx, dy, dz = (0, 0, len(coord_map))
+
+    for xy in coord_map:
+        if len(xy) > dx:
+            dy = len(xy)
+        for x in xy:
+            if len(x) > dx:
+                dx = len(x)
+            num += len(x)
+
+    result = [None] * num
+    for z in range(dz):
+        for y in range(dy):
+            for x in range(dx):
+                result[coord_map[z][y][x]] = [x, y, z]
+    return result
+
+
+def calc_ring_pixel_count(rings):
+    num = 0
+    for r in rings:
+        num += len(r)
+    return num
+
+
+def calc_ring_steps(rings):

```

Comment:
ok. ok... but I don't want to have to remember that one liner every where I need it which is why I wrapped it in a function. Or you just saying change the contents of the function?